spec/models/sponsor_spec.rb

```diff
@@ -29,13 +29,37 @@
   it { is_expected.to validate_inclusion_of(:payment_plan).in_array (Sponsor::PAYMENT_PLANS << '') }
   it { is_expected.to validate_inclusion_of(:country).in_array ISO3166::Country.countries.map { |c| c[1] } - ['IL'] }
 
-  [7, 'yes', true].each do |bad_date_value|
-    it { is_expected.to_not allow_value(bad_date_value).for :start_date }
-  end
 
   it { is_expected.to validate_numericality_of(:requested_orphan_count).
                           only_integer.is_greater_than(0) }
 
+  context 'start_date validation' do
+    describe 'valid date' do
+      it { is_expected.to have_validation :valid_date_presence, :on => :start_date }
+    end
+
+    describe 'not beyond OSRA establishment date' do
+      it { is_expected.to have_validation :date_beyond_osra_establishment, :on => :start_date }
+    end
+
+    describe "date_not_beyond_first_of_next_month" do
+      past = 1.year.ago
+      today = Date.current
+      first_of_next_month = today.beginning_of_month.next_month
+      yesterday = today.yesterday
+      second_of_next_month = first_of_next_month + 1.day
+      two_months_ahead = today + 2.months
+
+      [past, today, first_of_next_month, yesterday].each do |good_date|
+        it { is_expected.to allow_value(good_date).for :start_date }
+      end
+
+      [second_of_next_month, two_months_ahead].each do |bad_date|
+        it { is_expected.to_not allow_value(bad_date).for :start_date }
+      end
+    end
+  end
+
   it { is_expected.to allow_value(nil, '', 'admin@example.com', 'some.email@192.168.100.100', 'grüner@grü.üne',
```

Comment:
This one too? I would be in favour of taking out everything that is not directly relevant to the core business logic of a model.